packages/component/src/provider.ts

```diff
@@ -99,11 +99,16 @@ export class Provider<T extends Name> {
     identifier?: string;
     optional?: boolean;
   }): NameServiceMapping[T] | null {
-    const { identifier, optional } = {
+    let { identifier, optional } = {
       identifier: DEFAULT_ENTRY_NAME,
       optional: false,
       ...options
     };
+
+    if (!identifier) {
```

Comment:
Is line 102 not working here, does an undefined/null identifier in the `...options` spread overwrite the default in 103? If so, maybe it's better to be explicit just using this `if` block here for identifier, and a similar block for `optional`, instead of the spread extend magic? Up to you though.

Reply:
updated with the suggestion.